python-checks/src/test/resources/checks/exceptionCauseType.py

```diff
@@ -0,0 +1,43 @@
+class NotAnException:
+    pass
+
+class MyException(Exception):
+    pass
+
+def raise_without_from():
+    raise TypeError()
+
+def raise_from_exception():
+    raise TypeError() from MyException()
+
+def raise_from_string():
+    raise TypeError() from "just a string" # Noncompliant
+
+def raise_from_non_exception():
+    raise TypeError() from NotAnException() # Noncompliant
+
+def raise_from_none():
+    raise TypeError() from None
+
+def raise_from_unknown(e):
+    raise TypeError() from e
+
+def assign_cause_to_exception(e):
+    e.__cause__ = MyException()
+
+def assign_cause_to_string(e):
+    e.__cause__ = "just a string" # Noncompliant
+
+def assign_cause_to_non_exception(e):
+    e.__cause__ = NotAnException() # Noncompliant
+
+def assign_cause_to_none(e):
+    e.__cause__ = None
+
+def assign_cause_to_unknown(e):
+    e.__cause__ = Unknown()
+
+def other_assignments(e):
+    e.__eq__ = NotAnException()
+    e = NotAnException()
+    e.__cause__ = f = NotAnException() # Noncompliant
```

Comment:
What about having an additional test such as:
```
def exception_tuple():
  try:
    raise ValueError()
  except (ValueError, TypeError) as e:
    raise MyException() from e
```
As I guess it would be a bit more challenging on our type inference? Currently it seems `e` resolves as "ANY" type, but we might want to ensure the behaviour is still correct when it resolves as an union type of ValueError, TypeError (as I believe it should?).

Reply:
Done in 425e643.